src/test/java/javax/jmdns/impl/DNSRecordTest.java

```diff
@@ -118,12 +117,12 @@ public void testServiceInfoFromDNSRecord() {
     @Test
     public void testDNSRecordKey() {
         DNSRecord record1 = new DNSRecord.Service("android_123.local.", DNSRecordClass.CLASS_IN, true, TTL_IN_SECONDS, 0, 0, 0, "test_server");
-        Assert.assertEquals("_android_123.local.", record1.getKey());
+        Assert.assertEquals("android_123.local.", record1.getKey());
```

Comment:
To be honest, I'm biased, because I like `mockito` ... but this was the ~point~ _underline_ (🥲) when I was unsure.. unsure whether the test is wrong here or the product. :)

Reply:
@carlonzo  you wrote this test and it used to have "_android" but I think the change here just broke that: https://github.com/jmdns/jmdns/pull/275 from @joerg1985

Reply:
actually I don't see why this ever should have had and "_" to start the name.

Reply:
Hey thanks for tagging, I'm pretty sure I've seen that coming from an mdns server on Android and was causing issues.

Reply:
@melloware 

What I would do then, is to put back the original `Assert` as it was before

```suggestion
        Assert.assertEquals("_android_123.local.", record1.getKey());
```

but modify the line above to create the record with a leading underscore:

```
... DNSRecord.Service("_android_123.local." ...
```
To check whether that _"special"_ leading character is not removed.

Reply:
OK test updated for underscore